### PR TITLE
wip : #10146 send an email when saving LogisticalForm if config enabled

### PR DESCRIPTION
**Pull request by dda-axelor**

### CODE REVIEW

File: axelor-stock/src/main/resources/domains/StockConfig.xml [hunk 1]
```diff
@@ -25,6 +25,10 @@
     <boolean name="realStockMoveAutomaticMail" title="Send email on stock move realization" default="false"/>
     <many-to-one name="plannedStockMoveMessageTemplate" title="Message template" ref="com.axelor.apps.message.db.Template"/>
     <many-to-one name="realStockMoveMessageTemplate" title="Message template" ref="com.axelor.apps.message.db.Template"/>
+
+    <boolean name="logisticalFormAutomaticMail" title="Send email when saving logistical form" default="false"/>
```
Comment (gdu-axelor) [inline axelor-stock/src/main/resources/domains/StockConfig.xml]: Email instead of mail

Comment (dda-axelor) [reply axelor-stock/src/main/resources/domains/StockConfig.xml]: done

File: axelor-stock/src/main/java/com/axelor/apps/stock/db/repo/LogisticalFormStockRepository.java [hunk 2]
```diff
@@ -34,19 +39,40 @@
...
+      Company company = logisticalForm.getCompany();
+
+      if (company != null) {
+        if (Strings.isNullOrEmpty(logisticalForm.getDeliveryNumber())) {
+          String sequenceNumber =
+              Beans.get(SequenceService.class)
+                  .getSequenceNumber("logisticalForm", logisticalForm.getCompany());
+          if (Strings.isNullOrEmpty(sequenceNumber)) {
+            throw new AxelorException(
+                Sequence.class,
```
Comment (ale-axelor) [inline axelor-stock/src/main/java/com/axelor/apps/stock/db/repo/LogisticalFormStockRepository.java]: Please use select value instead of string literal.

Comment (dda-axelor) [reply axelor-stock/src/main/java/com/axelor/apps/stock/db/repo/LogisticalFormStockRepository.java]: done

**Pull request merged**